core_planning/twist_gate/src/twist_gate.cpp

```diff
@@ -28,66 +28,67 @@
  *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
-#include "twist_gate/twist_gate.h"
-#include <ros_observer/lib_ros_observer.h>
+#include "twist_gate/twist_gate.hpp"
 
 #include <memory>
 #include <string>
-#include <autoware_system_msgs/DiagnosticStatus.h>
 
-using AwDiagStatus = autoware_system_msgs::DiagnosticStatus;
+namespace std_ph = std::placeholders;
 
-TwistGate::TwistGate(const ros::NodeHandle& nh, const ros::NodeHandle& private_nh)
-  : nh_(nh)
-  , private_nh_(private_nh)
-  , timeout_period_(10.0)
+TwistGate::TwistGate(const rclcpp::NodeOptions &options)
+    : carma_ros2_utils::CarmaLifecycleNode(options), timeout_period_(10,0)
 {
-  private_nh_.param<bool>("use_decision_maker", use_decision_maker_, false);
-  private_nh_.param<bool>("use_lgsim", use_lgsim_, false);
-  private_nh_.param<bool>("use_twist", use_twist_, false);
-
-  health_checker_ptr_ = std::make_shared<autoware_health_checker::HealthChecker>(nh_, private_nh_);
-  control_command_pub_ = nh_.advertise<std_msgs::String>("/ctrl_mode", 1);
-  vehicle_cmd_pub_ = nh_.advertise<vehicle_cmd_msg_t>("/vehicle_cmd", 1, true);
-  config_sub_ = nh_.subscribe("config/twist_filter", 1, &TwistGate::configCallback, this);
-
-  auto_cmd_sub_stdmap_["twist_cmd"] = nh_.subscribe("/twist_cmd", 1, &TwistGate::twistCmdCallback, this);
-  auto_cmd_sub_stdmap_["ctrl_cmd"] = nh_.subscribe("/ctrl_cmd", 1, &TwistGate::ctrlCmdCallback, this);
-  auto_cmd_sub_stdmap_["lamp_cmd"] = nh_.subscribe("/lamp_cmd", 1, &TwistGate::lampCmdCallback, this);
-  auto_cmd_sub_stdmap_["state"] = nh_.subscribe("/decision_maker/state", 1, &TwistGate::stateCallback, this);
-  auto_cmd_sub_stdmap_["emergency_velocity"] =
-      nh_.subscribe("emergency_velocity", 1, &TwistGate::emergencyCmdCallback, this);
-
-  output_msg_.header.seq = 0;
-  emergency_stop_msg_.data = false;
-  health_checker_ptr_->ENABLE();
-  health_checker_ptr_->NODE_ACTIVATE();
-
-  emergency_handling_time_ = ros::Time::now();
+
+    //Declare parameters
+    use_decision_maker_ = declare_parameter<bool>("~use_decision_maker", use_decision_maker_);
+    use_lgsim_ = declare_parameter<bool>("use_lgsim", use_lgsim_);
+    use_twist_ = declare_parameter<bool>("use_twist", use_twist_);
+
 }
 
-void TwistGate::twistCmdCallback(const geometry_msgs::TwistStamped::ConstPtr& input_msg)
+carma_ros2_utils::CallbackReturn TwistGate::handle_on_configure(const rclcpp_lifecycle::State &prev_state)
 {
+    //get parameters
+    get_parameter<bool>("use_decision_maker", use_decision_maker_);
+    get_parameter<bool>("use_lgsim", use_lgsim_);
+    get_parameter<bool>("use_twist", use_twist_);
+    
+    // TODO: control_command_pub_ is unused
```

Comment:
Yes, this TODO was added in this PR because control_command_pub is not being used. Since it was part of the ros1 implementation it hasn't been removed